tools/scripts/list_stl_dependencies.sh

```diff
@@ -22,30 +22,73 @@
 #   ./list_stl_dependencies.sh
 #=====================================================
 
+SCOPE=${1:-list}
 COMPONENTS=(iceoryx_hoofs iceoryx_posh)
 SOURCE_DIR=(source include)
 WORKSPACE=$(git rev-parse --show-toplevel)
+QNX_PLATFORM_DIR=$WORKSPACE/iceoryx_hoofs/platform/qnx/
+WHITELIST=$WORKSPACE/tools/scripts/header-allowlist.txt
+TEMP_FILE=$(mktemp)
 
 for COMPONENT in ${COMPONENTS[@]}; do
     for DIR in ${SOURCE_DIR[@]}; do
         GREP_PATH="${GREP_PATH} ${WORKSPACE}/${COMPONENT}/$DIR"
     done
 done
 
+echo "# QNX platform/ libc headers" >> $TEMP_FILE
+
+echo "# usage of headers for QNX"
+gcc -w -fpreprocessed -dD -E $(find $QNX_PLATFORM_DIR -type f -iname *.cpp -o -iname *.hpp) \
```

Comment:
I agree with you that this would be definitely a nicer solution! However, as `gcc`s preprocessing stage is invoked here and it e.g. doesn't handle `.inl` and `.in`, that's the reason why `.inl`s are currently passed unprocessed with comments, I would argue to go with common sense here and not consider extensions like `.cxx`, `.cc`, etc. as they are not used in iceoryx yet. Could you go along with that?

Reply:
Could you then also please check the headers from `build/generated` or even better the headers which are installed? This would be the best but then the script would need to run on QNX

Reply:
Event though we currently don't have `.inl` files in the platform folder, please make it future proof nad handle it exactly like below with the header from iceoryx_hoofs and iceoryx_posh. These things tend to break when one is short on time.